kubetest/anywhere.go

```diff
@@ -199,3 +205,126 @@ func (k kubernetesAnywhere) Down() error {
 	}
 	return finishRunning(exec.Command("make", "-C", k.path, "FORCE_DESTROY=y", "destroy"))
 }
+
+const DefaultConfigFile = ".config"
+
+type kubernetesAnywhereMulti struct {
+	*kubernetesAnywhere
+	multiClusters multiClusterDeployment
+}
+
+var _ deployer = kubernetesAnywhereMulti{}
+
+func newKubernetesAnywhereMulti(project, zone string, multiClusters multiClusterDeployment) (*kubernetesAnywhereMulti, error) {
+	if len(multiClusters.clusters) < 1 {
+		return nil, fmt.Errorf("invalid --multi-clusters flag passed")
+	}
+	k, _ := newKubernetesAnywhere(project, zone)
+	mk := &kubernetesAnywhereMulti{k, multiClusters}
+
+	for _, cluster := range mk.multiClusters.clusters {
+		specificZone, specified := mk.multiClusters.zones[cluster]
+		if specified {
+			mk.Zone = specificZone
+		} else {
+			mk.Zone = zone
+		}
+		mk.Cluster = cluster
+		mk.KubeContext = mk.Zone + "-" + mk.Cluster
+		if err := mk.writeConfig(); err != nil {
+			return nil, err
+		}
+	}
+	return mk, nil
+}
+
+func (k kubernetesAnywhereMulti) writeConfig() error {
+	config, err := k.getConfig()
+	if err != nil {
+		return fmt.Errorf("Could not generate config: %v", err)
+	}
+
+	configFileName := DefaultConfigFile + "-" + k.Cluster
+	f, err := os.Create(k.path + "/" + configFileName)
+	if err != nil {
+		return fmt.Errorf("Could not create file: %v", err)
+	}
+	defer f.Close()
+
+	fmt.Fprint(f, config)
+	return nil
+}
+
+func (k kubernetesAnywhereMulti) Up() error {
+	var cmds []*exec.Cmd
+	for _, cluster := range k.multiClusters.clusters {
+		configFile := DefaultConfigFile + "-" + cluster
```

Comment:
This concatenation to create the config file name is duplicated enough that I would just save it in a field of `kubernetesAnywhereMulti`.

Reply:
okie, done as suggested.